File: algorithm study/BOJ/1. Bonze/2747.js
// boj 2747 피보나치 수
// recursion

const input = require("fs").readFileSync("./example.txt").toString().split("\n");
// const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const n = input[0].split(" ").map(Number);

let arr = [];
for (let i = 0; i <= n; i++) {
  if (i < 2) arr[i] = i;
  else arr[i] = arr[i - 1] + arr[i - 2];
}
console.log(arr[n]);

File: algorithm study/BOJ/1. Bonze/2747.cpp
// boj 2747 피보나치 수
// recursion
#include <iostream>
using namespace std;
int arr[46];
int n;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    cin >> n;
    for (int i = 0; i <= n; i++) {
        if (i < 2)
            arr[i] = i;
        else
            arr[i] = arr[i - 1] + arr[i - 2];
    }
    cout << arr[n];
    return 0;
}